x-pack/test/functional/apps/maps/import_geojson/add_layer_import_panel.js

```diff
@@ -0,0 +1,159 @@
+
+  describe('Import layer panel', () => {
+    before(async () => {
+      await PageObjects.maps.openNewMap();
+    });
+
+    beforeEach(async () => {
+      await PageObjects.maps.clickAddLayer();
+      const panelOpen = await PageObjects.maps.isLayerAddPanelOpen();
+      expect(panelOpen).to.be(true);
+
+      await PageObjects.maps.selectGeoJsonUploadSource();
+      await PageObjects.maps.uploadJsonFileForIndexing(
+        path.join(__dirname, FILE_LOAD_DIR, DEFAULT_LOAD_FILE_NAME)
+      );
+      await PageObjects.maps.waitForLayersToLoad();
+      const layerLoadedInToc = await PageObjects.maps
+        .doesLayerExist(IMPORT_FILE_PREVIEW_NAME);
+      expect(layerLoadedInToc).to.be(true);
+
+      const filePickerLoadedFile = await PageObjects.maps
+        .hasFilePickerLoadedFile(DEFAULT_LOAD_FILE_NAME);
+      expect(filePickerLoadedFile).to.be(true);
+    });
+
+    afterEach(async () => {
+      await PageObjects.maps.cancelLayerAdd();
+    });
+
+    it('should close & remove preview layer on clicking "Cancel" after uploading file',
+      async () => {
+        await PageObjects.maps.cancelLayerAdd();
+        const panelOpen = await PageObjects.maps.isLayerAddPanelOpen();
+        expect(panelOpen).to.be(false);
+
+        await PageObjects.maps.waitForLayerDeleted(IMPORT_FILE_PREVIEW_NAME);
+        const layerLoadedInToc = await PageObjects.maps.doesLayerExist(IMPORT_FILE_PREVIEW_NAME);
+        expect(layerLoadedInToc).to.be(false);
+      });
+
+    it('should replace layer on input change',
+      async () => {
+        // Upload second file
+        const secondLoadFileName = 'polygon.json';
+        await PageObjects.maps.uploadJsonFileForIndexing(
+          path.join(__dirname, FILE_LOAD_DIR, secondLoadFileName)
+        );
+        await PageObjects.maps.waitForLayersToLoad();
+        // Check second file is loaded in file picker
+        const filePickerLoadedFile = await PageObjects.maps
+          .hasFilePickerLoadedFile(secondLoadFileName);
+        expect(filePickerLoadedFile).to.be(true);
+      });
+
+    it('should clear layer on replacement layer load error',
+      async () => {
+        // Upload second file
+        const secondLoadFileName = 'not_json.txt';
+        await PageObjects.maps.uploadJsonFileForIndexing(
+          path.join(__dirname, FILE_LOAD_DIR, secondLoadFileName)
+        );
+        await PageObjects.maps.waitForLayersToLoad();
+        // Check second file is loaded in file picker
+        const filePickerLoadedFile = await PageObjects.maps
+          .hasFilePickerLoadedFile(secondLoadFileName);
+        expect(filePickerLoadedFile).to.be(true);
+        // Check that no file is loaded in layer preview
+        const layerLoadedInToc = await PageObjects.maps.doesLayerExist(IMPORT_FILE_PREVIEW_NAME);
+        expect(layerLoadedInToc).to.be(false);
+      });
+
+    it('should supply the correct index type(s) (geo_point or geo_shape) for the source',
```

Comment:
Does this really need to be tested in a functional test? Seems like a unit test can cover this case sufficiently since there is not a lot of play across components for this feature

Reply:
Taking a fresh look, I would say yeah. I'll remove it